### DISCUSSION TITLE
Introducing Hexagon arch plugin

### DISCUSSION THREAD

**Discussion by cfircohen** [Vector35/binaryninja-api#2221] [Show and tell] [answered]
Hello everyone!
I've just released my Hexagon arch plugin: https://github.com/google/binja-hexagon

It has a complete disassembler support:
![image](https://user-images.githubusercontent.com/17216172/103834622-c517e400-5038-11eb-821b-ee09e4d7009a.png)

and (partial) LLIL lifting support:
![image](https://user-images.githubusercontent.com/17216172/103834664-e7a9fd00-5038-11eb-9217-3259117502b8.png)

I wanted to stop by, and **thank you** for building Binary Ninja! It's such an awesome product. I've had a good experience building this plugin using your API. I found the LLIL language design very clean and intuitive, and overall, things just worked.

Cheers,
Cfir

**Comment by psifertex** [comment, marked as answer]
Thanks so much for the contribution! We were pleasantly surprised to see this released. Appreciate your hard work, and that feedback is much appreciated. 

You might be interested in joining the #hexagon channel in our public slack (https://slack.binary.ninja/) as there are several other people who've made various amounts of progress implementing hexagon analysis modules on BN. It's definitely got some challenging aspects!

**Reply by psifertex** [reply]
Oh, one other thing I wanted to mention -- right our our plugin manager doesn't work well with native plugins that require pre-built binaries which is something we're looking into improving since we'd love to make this easier to install. As one of the few public native plugins, we'll definitely reach out when we start to work on that.
